falcon/request.py

```diff
@@ -1075,6 +1075,7 @@ def _parse_form_urlencoded(self):
         # content length will only ever be read once per
         # request in most cases.
         body = self.stream.read()
+        self.stream.seek(0)
```

Comment:
OK, I see that `seek` is not part of the [spec](https://www.python.org/dev/peps/pep-3333/#input-and-error-streams). I think the BytesIO suggested in #493 is the best compromise to move forward at this point.